.changeset/curly-ghosts-pull.md

```diff
@@ -0,0 +1,20 @@
+---
+"@paprika/input": major
+---
+
+### Added
+
+- Added an imperative `focus()` method.
+- Added a props collector component, `<Input.Container>`, to apply attributes to root `<div>` element.
+
+### Removed
+
+- Removed `className` prop.
+- Removed all classNames from DOM elements.
+
+### Migration Guide
+
+- If a custom class name is required on the root element, it can be added like `<Input><Input.Container className="custom"></Input>`.
+- If classNames were used to target the `<input>`, they will need to be udpated to use attributes like `data-pka-anchor`, `aria-invalid`, etc.
```

Comment:
so now doing: `<Input className="my-custom" />` should add a classname in the input HTML element?

Reply:
Yes, exactly! It _used to_ go on the root `<div>` but now it would go on the `<input>`.  

If you had added any other kind of attribute though, like `<Input data-testid>` it _used to_ to on the `<input>` and it will _still_ go on the `<input>`. 

The change is that you _can_ put that `data-testid` attribute on the root `<div>` if you need to now.